Mathlib/Algebra/Module/Torsion.lean

```diff
@@ -911,8 +911,16 @@ variable (A : Type*) [AddCommGroup A] (n : ℤ)
 def torsionBy : AddSubgroup A :=
   (Submodule.torsionBy ℤ A n).toAddSubgroup
 
-@[inherit_doc]
-scoped notation:max (priority := high) A"["n"]" => torsionBy A n
+@[inherit_doc torsionBy]
+scoped syntax:max (name := torsionByStx) (priority := high) term noWs "[" term "]" : term
+
+macro_rules | `($A[$n]) => `(torsionBy $A $n)
+
+/-- Unexpander for `torsionBy`. -/
+@[app_unexpander torsionBy]
```

Comment:
I think this and the other should be scoped, since otherwise I believe this will cause it to pretty print all the time.
```suggestion
@[scoped app_unexpander torsionBy]
```
I think there's no harm in `macro_rules` not being `scoped macro_rules`; it might be better this way because if some other macro comes along using the notation it won't break just because the macro rule isn't opened.

Reply:
Thanks for catching this!